server/sandbox_remove.go

```diff
@@ -81,6 +81,10 @@ func (s *Server) RemovePodSandbox(ctx context.Context, req *pb.RemovePodSandboxR
 		return nil, fmt.Errorf("failed to remove networking namespace for sandbox %s: %v", sb.id, err)
 	}
 
+	s.removeContainer(podInfraContainer)
+	sb.infraContainer = nil
+	s.removeSandbox(sb.id)
```

Comment:
Just this line should be moved after s.ctrIDIndex.Delete. Otherwise looks good. Thanks!